src/Foreign/JavaScript/Types.hs

```diff
@@ -265,6 +267,7 @@ data Window = Window
 
     , wCallBuffer     :: TVar (String -> String)
     , wCallBufferMode :: TVar CallBufferMode
+    , wCallBufferStats :: TMVar UTCTime
```

Comment:
I think the code can be clarified by using an ordinary `TVar` instead.
The single use of `takeMVar` can be rewritten by using the `retry` function.